public/app/features/explore/AddToDashboard/addToDashboard.ts

```diff
@@ -25,13 +39,28 @@ const createDashboard = (
   return getDashboardSrv().saveDashboard({ dashboard, folderId }, { showErrorAlert: false, showSuccessAlert: false });
 };
 
-export const addToDashboard = async (data: SaveToNewDashboardDTO, options: SaveOptions): Promise<string> => {
-  const res = await createDashboard(
-    data.dashboardName,
-    data.folderId,
-    options.queries,
-    options.datasource,
-    options.panel
-  );
-  return res.data.url;
+const addPanelToDashboard = async (uid: string, queries: DataQuery[], datasource: DataSourceRef, panel: string) => {
+  const dashboarData = await backendSrv.getDashboardByUid(uid);
+
+  const dashboard = new DashboardModel(dashboarData.dashboard, dashboarData.meta);
+
+  dashboard.addPanel({ targets: queries, type: panel, title: 'New Panel', datasource });
+
+  return getDashboardSrv().saveDashboard({ dashboard }, { showErrorAlert: false, showSuccessAlert: false });
+};
```

Comment:
This feels like very hit and miss, do add and save the dashboard from explore. Feels like this will 100% time require the user to later find and open this dashboard to correct the layout. 

I think it would be better to copy the panel to the clipboard and use the "Paste panel from clipboard" feature:

![Screenshot from 2022-03-16 14-50-31](https://user-images.githubusercontent.com/10999/158604890-2c313ddd-5c00-4e89-b6fc-22a5c5e1e73b.png)

The panel json is not actually in the clipboard but in local storage I think. This way the user can use the normal add panel view to see the panel added, and place it / correct layout, and THEN save the dashboard. 

To make the experiance better I think the user would pick a dashboard to add it to, this would open that dashbord with a query param (or some other redux state set) so we know we come from explore "add panel" flow and automatically trigger the add panel flow. But not save the dashboard. 

We do the same when you want to go back to the dashboard with your changes, we don't save from explore, we just open the dashboard and apply the changes, and let the user save the dashboard like normal.

Reply:
Hey Torkel,
thanks for your feedback!
We conducted exploratory research and 10 usability tests with Grafanistas and external participants regarding this Explore-to-dashboard workflow. Let me know if you want me to set up a time to share the whole research with you.

Reply:
Yes, please do. Please saving directly from explore without first seeing and adjusting the dashboard would leave all layouts broken, so find it hard to believe that was the preferred choice